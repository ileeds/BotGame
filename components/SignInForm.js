import React, { Component } from 'react';
import { View } from 'react-native';
import { FormLabel, FormInput, Button } from 'react-native-elements';
import { connect } from 'react-redux';
import { sendCode, anotherCode } from '../actions';
import { renderErrorAndLoading } from '../appUtils/renderFunctions';
import NavigationService from '../appUtils/NavigationService';

class SignInForm extends Component {
	state = { code: '' };

	render() {
		return (
			<View>
				{renderErrorAndLoading(this.props.error, this.props.loading)}
				<View style={{ marginBottom: 10 }}>
					<FormLabel>Enter Code</FormLabel>
					<FormInput
						value={this.state.code}
						onChangeText={code => this.setState({ code })}
						keyboardType={'phone-pad'}
					/>
				</View>

				<Button
					onPress={() => this.props.sendCode(this.state.code)}
					title="Submit"
				/>

				<Button
					onPress={() => NavigationService.navigate('getCode')}
					title="Get Another Code"
				/>
			</View>
		);
	}
}

const mapStateToProps = ({ auth }) => {
	const { error, loading } = auth;
	return { error, loading };
};

export default connect(
	mapStateToProps,
	{ sendCode }
)(SignInForm);
