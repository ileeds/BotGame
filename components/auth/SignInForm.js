import React, { Component } from "react";
import { View } from "react-native";
import { connect } from "react-redux";
import CodeInput from "react-native-confirmation-code-input";
import { sendCode } from "../../actions";
import { renderErrorAndLoading } from "../../appUtils/renderFunctions";
import NavigationService from "../../appUtils/NavigationService";
import ButtonLarge from "../ButtonLarge";
import { CardSection } from "../common";
import { backgroundColor } from "../../appUtils/puppet";

export class SignInForm extends Component {
  render() {
    return (
      <View id="SignInForm" style={{ backgroundColor, flex: 1 }}>
        <View style={{ flex: 0.5 }} />
        <CardSection>
          <CodeInput
            id="CodeInput"
            ref="codeInputRef"
            codeLength={4}
            space={20}
            size={50}
            inputPosition="center"
            keyboardType="numeric"
            onFulfill={code => this.props.sendCode(code)}
          />
        </CardSection>

        <CardSection>
          <ButtonLarge
            id="ButtonLarge"
            text="Get Another Code"
            onPress={() => {
              this.refs.codeInputRef.clear();
              NavigationService.navigate("getCode");
            }}
          />
        </CardSection>

        {renderErrorAndLoading(this.props.error, this.props.loading)}
        <View style={{ flex: 2 }} />
      </View>
    );
  }
}

const mapStateToProps = ({ auth }) => {
  const { error, loading } = auth;
  return { error, loading };
};

export default connect(
  mapStateToProps,
  { sendCode }
)(SignInForm);
