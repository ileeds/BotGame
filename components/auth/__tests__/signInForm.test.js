import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";
import { SignInForm } from "../SignInForm";

let props, wrapper;

const initialState = {
  auth: {
    error: "",
    loading: false
  }
};

const mockStore = configureStore();
const store = mockStore(initialState);

beforeEach(() => {
  props = {
    sendCode: jest.fn()
  };

  wrapper = shallow(<SignInForm {...props} />, {
    context: { store: mockStore(initialState) }
  });
});

it("renders correctly", () => {
  expect(wrapper.dive()).toMatchSnapshot();
});

describe("SignInForm", () => {
  it("should render self and subcomponents", () => {
    expect(wrapper.find("#SignInForm").length).toBe(1);
    expect(wrapper.find("#CodeInput").length).toBe(1);
    expect(wrapper.find("#ButtonLarge").length).toBe(1);
  });

  it("should call sendCode on code input fulfill", () => {
    const input = wrapper.find("#CodeInput");
    input.props().onFulfill(1111);
    expect(props.sendCode.mock.calls.length).toBe(1);
  });
});
