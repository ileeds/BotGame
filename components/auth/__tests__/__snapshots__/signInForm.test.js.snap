// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component
    id="SignInForm"
    style={
      Object {
        "backgroundColor": "#3E5382",
        "flex": 1,
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 0.5,
        }
      }
    />
    <CardSection>
      <ConfirmationCodeInput
        activeColor="rgba(255, 255, 255, 1)"
        autoFocus={true}
        cellBorderWidth={1}
        className="border-box"
        codeLength={4}
        compareWithCode=""
        id="CodeInput"
        ignoreCase={false}
        inactiveColor="rgba(255, 255, 255, 0.2)"
        inputPosition="center"
        keyboardType="numeric"
        onFulfill={[Function]}
        size={50}
        space={20}
      />
    </CardSection>
    <CardSection>
      <ButtonLarge
        id="ButtonLarge"
        onPress={[Function]}
        text="Get Another Code"
      />
    </CardSection>
    <Component
      style={
        Object {
          "flex": 2,
        }
      }
    />
  </Component>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "flex": 0.5,
            }
          }
        />,
        <CardSection>
          <ConfirmationCodeInput
            activeColor="rgba(255, 255, 255, 1)"
            autoFocus={true}
            cellBorderWidth={1}
            className="border-box"
            codeLength={4}
            compareWithCode=""
            id="CodeInput"
            ignoreCase={false}
            inactiveColor="rgba(255, 255, 255, 0.2)"
            inputPosition="center"
            keyboardType="numeric"
            onFulfill={[Function]}
            size={50}
            space={20}
          />
        </CardSection>,
        <CardSection>
          <ButtonLarge
            id="ButtonLarge"
            onPress={[Function]}
            text="Get Another Code"
          />
        </CardSection>,
        undefined,
        <Component
          style={
            Object {
              "flex": 2,
            }
          }
        />,
      ],
      "id": "SignInForm",
      "style": Object {
        "backgroundColor": "#3E5382",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "flex": 0.5,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ConfirmationCodeInput
            activeColor="rgba(255, 255, 255, 1)"
            autoFocus={true}
            cellBorderWidth={1}
            className="border-box"
            codeLength={4}
            compareWithCode=""
            id="CodeInput"
            ignoreCase={false}
            inactiveColor="rgba(255, 255, 255, 0.2)"
            inputPosition="center"
            keyboardType="numeric"
            onFulfill={[Function]}
            size={50}
            space={20}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeColor": "rgba(255, 255, 255, 1)",
            "autoFocus": true,
            "cellBorderWidth": 1,
            "className": "border-box",
            "codeLength": 4,
            "compareWithCode": "",
            "id": "CodeInput",
            "ignoreCase": false,
            "inactiveColor": "rgba(255, 255, 255, 0.2)",
            "inputPosition": "center",
            "keyboardType": "numeric",
            "onFulfill": [Function],
            "size": 50,
            "space": 20,
          },
          "ref": "codeInputRef",
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ButtonLarge
            id="ButtonLarge"
            onPress={[Function]}
            text="Get Another Code"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "ButtonLarge",
            "onPress": [Function],
            "text": "Get Another Code",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "flex": 2,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "flex": 0.5,
              }
            }
          />,
          <CardSection>
            <ConfirmationCodeInput
              activeColor="rgba(255, 255, 255, 1)"
              autoFocus={true}
              cellBorderWidth={1}
              className="border-box"
              codeLength={4}
              compareWithCode=""
              id="CodeInput"
              ignoreCase={false}
              inactiveColor="rgba(255, 255, 255, 0.2)"
              inputPosition="center"
              keyboardType="numeric"
              onFulfill={[Function]}
              size={50}
              space={20}
            />
          </CardSection>,
          <CardSection>
            <ButtonLarge
              id="ButtonLarge"
              onPress={[Function]}
              text="Get Another Code"
            />
          </CardSection>,
          undefined,
          <Component
            style={
              Object {
                "flex": 2,
              }
            }
          />,
        ],
        "id": "SignInForm",
        "style": Object {
          "backgroundColor": "#3E5382",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "flex": 0.5,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ConfirmationCodeInput
              activeColor="rgba(255, 255, 255, 1)"
              autoFocus={true}
              cellBorderWidth={1}
              className="border-box"
              codeLength={4}
              compareWithCode=""
              id="CodeInput"
              ignoreCase={false}
              inactiveColor="rgba(255, 255, 255, 0.2)"
              inputPosition="center"
              keyboardType="numeric"
              onFulfill={[Function]}
              size={50}
              space={20}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeColor": "rgba(255, 255, 255, 1)",
              "autoFocus": true,
              "cellBorderWidth": 1,
              "className": "border-box",
              "codeLength": 4,
              "compareWithCode": "",
              "id": "CodeInput",
              "ignoreCase": false,
              "inactiveColor": "rgba(255, 255, 255, 0.2)",
              "inputPosition": "center",
              "keyboardType": "numeric",
              "onFulfill": [Function],
              "size": 50,
              "space": 20,
            },
            "ref": "codeInputRef",
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ButtonLarge
              id="ButtonLarge"
              onPress={[Function]}
              text="Get Another Code"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "ButtonLarge",
              "onPress": [Function],
              "text": "Get Another Code",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "flex": 2,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
